File: Clase__75__15-10__Room/EjercicioRoom/app/src/main/java/com/example/ejercicioroom/fragmentos/ListFragment.java
package com.example.ejercicioroom.fragmentos;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.ejercicioroom.R;
import com.example.ejercicioroom.adapter.CustomViewHolder;
import com.example.ejercicioroom.adapter.ProductoAdapter;
import com.example.ejercicioroom.adapter.ProductoListAdapter;
import com.example.ejercicioroom.databinding.FragmentAddBinding;
import com.example.ejercicioroom.databinding.FragmentListBinding;
import com.example.ejercicioroom.model.Producto;
import com.example.ejercicioroom.model.ProductoEntity;
import com.example.ejercicioroom.viewmodel.ProductoViewModel;

import java.util.ArrayList;


public class ListFragment extends Fragment {
    FragmentListBinding binding;
    private ProductoViewModel viewModel;
    private RecyclerView recyclerView;
    private ProductoListAdapter adapter;



    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentListBinding.inflate(inflater, container, false);
        View view = binding.getRoot();

        viewModel = new ViewModelProvider(this).get(ProductoViewModel.class);

        recyclerView = binding.recycler;

        adapter = new ProductoListAdapter(new ProductoListAdapter.ProductoDiff());
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerView.setAdapter(adapter);



        adapter.setListener(productoEntity -> {


            viewModel.getCurrentProduct().postValue(adapter.getProductoEntity());
            Log.i("prueba", productoEntity.getMarca().toString());
            //Navigation.findNavController(view).navigate(R.id.detailsFragment);
        });

        viewModel.selectAllProductos().observe(getViewLifecycleOwner(), productos ->{
            adapter.submitList(productos);
        } );

//viewModel.getCurrentProduct().setValue(adapter.getItemId(position));

//        recyclerView.addOnItemTouchListener(l->{
//
//        });

//        viewModel.setCurrentProduct(
//
//                MutableLiveData<adapter.getCurrentProduct()>);
        //binding.recycler.findViewHolderForAdapterPosition()

        //ProductoAdapter adapter = new ProductoAdapter(listadoProductos);





        return view;


    }
}

File: Clase__84__28-10__Recycler&ViewModel/ContactosApp/app/src/main/java/com/example/simplecontactos/fragments/ui/ListFragment.kt
package com.example.simplecontactos.fragments.ui

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import androidx.fragment.app.activityViewModels
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.simplecontactos.R
import com.example.simplecontactos.adapter.ContactoAdapter
import com.example.simplecontactos.databinding.FragmentListBinding
import com.example.simplecontactos.viewmodel.ContactoViewModel

class ListFragment : Fragment() {
   private lateinit var binding: FragmentListBinding

   val viewModel : ContactoViewModel by activityViewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
       binding = FragmentListBinding.inflate(layoutInflater, container, false)

        val adapter = ContactoAdapter()

        with(binding){
            recycler.layoutManager = LinearLayoutManager(context)
            recycler.adapter = adapter

        }

        viewModel.listaContacto.observe(viewLifecycleOwner, Observer {
            adapter.setContacto(it)
        })


        return binding.root

    }


}